tools/cdi-source-update-poller/main.go

```diff
@@ -58,6 +64,8 @@ func main() {
 	if err != nil {
 		log.Fatalf("Failed BuildConfigFromFlags, kubeURL %s configPath %s: %v", kubeURL, configPath, err)
 	}
+	cfg.TLSClientConfig = rest.TLSClientConfig{Insecure: true}
```

Comment:
hmm which change in your PR now requires this?
this is just about the cdi client that we use to get stuff (not the polling), why do we have to go insecure?

Reply:
The CronJob containerized poller app runs in CDI namespace, and it tries to Get/Update() DataImportCron which is in another namespace, so when proxy is configured, Get/Update() fail on certificate issues unless we configure TLS appropriately. Maybe `Insecure: true` is not the best idea here, but I have no idea how to use the certs dir for setting TLSClientConfig.

Reply:
AFAIK, under the hood, it should respect proxy env vars so not sure what goes wrong
https://github.com/kubevirt/containerized-data-importer/blob/206c77afdaad91e5be99460a76a1a5d3e5131841/vendor/k8s.io/client-go/rest/config.go#L137-L142

Reply:
ok, but what would you append?

Reply:
hmm if it was erroring on x509 I think
https://github.com/kubernetes/apiserver/blob/25ccbfa75c69642e44ac9be141e3bacc7edb3879/pkg/util/webhook/client.go#L136-L140 could work, or something nicer like placing the cert in the golang "respected" path https://go.dev/src/crypto/x509/root_linux.go

Reply:
Actually thinking about it a bit more, you don't need to proxy the k8s calls at all. So you don't have to mess with the `cfg` settings at all.

Reply:
I think they are being proxied because of the env vars, usually, we don't set the env vars on containers that do k8s API interactions

Reply:
Fixing that

Reply:
Nice and clean
Just making sure here: the reason we don't care about proxying the k8s API calls is that it's somehow taken care of in another internal k8s layer?

Reply:
We have no good reason to proxy the 2 k8s get/update calls here. Do we?

Reply:
No the assumption is that the proxy is on the edge of the cluster (aka stuff that tries to go outside of the cluster). I see no use case where it makes sense to proxy traffic inside the cluster.